src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] supplyAndBuy = readData(fromFileName);
+        int result = processStatistics(supplyAndBuy[0], supplyAndBuy[1]);
+        writeData(toFileName, supplyAndBuy[0], supplyAndBuy[1], result);
+    }
+
+    private int[] readData(String fromFileName) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
```

Comment:
Use constants for 'supply' and 'buy' strings to avoid magic values in the code and to make it easier to change and understand.